From 1bdbe778ac4d634974d150ee141eac45cb9aa777 Mon Sep 17 00:00:00 2001
From: Jernej Skrabec <jernej.skrabec@gmail.com>
Date: Sat, 23 May 2026 10:10:43 +0000
Subject: [PATCH] dt-bindings: display: Add A523 eDP/DP encoder binding

The Allwinner A523/A527/T527 family embeds an Innosilicon eDP 1.3 TX
controller, fed by TCON-TV1, driving either an internal eDP panel or
an external DisplayPort connector. Add a binding describing the two
variants and the corresponding clocks, reset and port graph.

Signed-off-by: Jernej Skrabec <jernej.skrabec@gmail.com>
---
 .../display/allwinner,sun55i-a523-edp.yaml    | 130 ++++++++++++++++++
 1 file changed, 130 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/display/allwinner,sun55i-a523-edp.yaml

diff --git a/Documentation/devicetree/bindings/display/allwinner,sun55i-a523-edp.yaml b/Documentation/devicetree/bindings/display/allwinner,sun55i-a523-edp.yaml
new file mode 100644
index 000000000000..b164fefab26d
--- /dev/null
+++ b/Documentation/devicetree/bindings/display/allwinner,sun55i-a523-edp.yaml
@@ -0,0 +1,130 @@
+# SPDX-License-Identifier: GPL-2.0-only OR BSD-2-Clause
+%YAML 1.2
+---
+$id: http://devicetree.org/schemas/display/allwinner,sun55i-a523-edp.yaml#
+$schema: http://devicetree.org/meta-schemas/core.yaml#
+
+title: Allwinner A523 eDP/DP TX Encoder
+
+maintainers:
+  - Jernej Skrabec <jernej.skrabec@gmail.com>
+
+description: |
+  The Allwinner A523/A527/T527 SoC family embeds an Innosilicon eDP 1.3
+  TX controller, fed by TCON-TV1. The controller exposes an AUX
+  channel, hot-plug detection and configurable link training, and can
+  drive either an internal eDP panel or an external DisplayPort
+  connector.
+
+properties:
+  compatible:
+    enum:
+      - allwinner,sun55i-a523-edp
+      - allwinner,sun55i-a523-dp
+
+  reg:
+    maxItems: 1
+
+  interrupts:
+    maxItems: 1
+
+  clocks:
+    items:
+      - description: Bus interface clock
+      - description: Module clock
+      - description: 24 MHz reference clock
+
+  clock-names:
+    items:
+      - const: bus
+      - const: mod
+      - const: 24m
+
+  resets:
+    maxItems: 1
+
+  reset-names:
+    const: bus
+
+  power-domains:
+    maxItems: 1
+
+  vdd-supply:
+    description: Controller core voltage rail.
+
+  vcc-supply:
+    description: I/O voltage rail powering the AUX/HPD pads.
+
+  ports:
+    $ref: /schemas/graph.yaml#/properties/ports
+
+    properties:
+      port@0:
+        $ref: /schemas/graph.yaml#/properties/port
+        description: Input endpoint, connected to TCON-TV1.
+
+      port@1:
+        $ref: /schemas/graph.yaml#/properties/port
+        description: |
+          Output endpoint, connected to an eDP panel or a DisplayPort
+          connector.
+
+    required:
+      - port@0
+      - port@1
+
+required:
+  - compatible
+  - reg
+  - interrupts
+  - clocks
+  - clock-names
+  - resets
+  - reset-names
+  - ports
+
+additionalProperties: false
+
+examples:
+  - |
+    #include <dt-bindings/interrupt-controller/arm-gic.h>
+
+    #define CLK_HDMI_24M	138
+    #define CLK_EDP		158
+    #define CLK_BUS_EDP	159
+    #define RST_BUS_EDP	72
+
+    edp@5720000 {
+        compatible = "allwinner,sun55i-a523-dp";
+        reg = <0x5720000 0x4000>;
+        interrupts = <GIC_SPI 144 IRQ_TYPE_LEVEL_HIGH>;
+        clocks = <&ccu CLK_BUS_EDP>,
+                 <&ccu CLK_EDP>,
+                 <&ccu CLK_HDMI_24M>;
+        clock-names = "bus", "mod", "24m";
+        resets = <&ccu RST_BUS_EDP>;
+        reset-names = "bus";
+
+        ports {
+            #address-cells = <1>;
+            #size-cells = <0>;
+
+            port@0 {
+                reg = <0>;
+
+                endpoint {
+                    remote-endpoint = <&tcon_tv1_out_edp>;
+                };
+            };
+
+            port@1 {
+                reg = <1>;
+
+                endpoint {
+                    remote-endpoint = <&dp_con_in>;
+                };
+            };
+        };
+    };
+
+...
-- 
2.53.0

